core/src/com/unciv/ui/worldscreen/Minimap.kt

```diff
@@ -120,13 +123,65 @@ class Minimap(val mapHolder: WorldMapHolder) : Table(){
 class MinimapHolder(mapHolder: WorldMapHolder): Table() {
     val minimap = Minimap(mapHolder)
     val worldScreen = mapHolder.worldScreen
+    private data class ToggleButtonInfo(val actor: Actor, val getIt: ()->Boolean, val setIt: (Boolean)->Unit)
+    private val toggleButtonInfo: EnumMap<MinimapToggleButtons,ToggleButtonInfo> = EnumMap<MinimapToggleButtons,ToggleButtonInfo>(MinimapToggleButtons::class.java)
 
     init {
         add(getToggleIcons()).align(Align.bottom)
         add(getWrappedMinimap())
         pack()
     }
 
+    enum class MinimapToggleButtons {
+        YIELD {
+            override val icon = "Food"
+        },
+        WORKED {
+            override val icon = "Population"
+        },
+        RESOURCES {
+            override val icon = "ResourceIcons/Cattle"
+        };
+        abstract val icon: String
+    }
+
+    // "Api" when external code wants to toggle something together with our buttons
+    fun getToggleButton(which: MinimapToggleButtons): Boolean {
+        val info = toggleButtonInfo[which] ?: return false
+        return info.getIt()
+    }
+    fun setToggleButton(which: MinimapToggleButtons, value: Boolean) {
+        val info = toggleButtonInfo[which] ?: return
+        info.setIt(value)
+        toggleButtonInfo[which]?.actor?.color?.a = if (value) 1f else 0.5f
+        worldScreen.shouldUpdate = true
+    }
+    fun syncToggleButton(which: MinimapToggleButtons) = setToggleButton(which,getToggleButton(which))
+    private fun flipToggleButton(which: MinimapToggleButtons) = setToggleButton(which,!getToggleButton(which))
+    private fun addToggleButton(table:Table, which: MinimapToggleButtons) {
+        val image = (
+                if ('/' in which.icon) {
+                    ImageGetter.getImage(which.icon)
+                        .surroundWithCircle(30f).apply { circle.color = Color.GREEN }
+                        .surroundWithCircle(40f, false)
+                } else {
+                    ImageGetter.getStatIcon(which.icon).surroundWithCircle(40f)
+                }
+            ).apply { circle.color = Color.BLACK }
```

Comment:
Please split this from the 'val image = if... else...'
it can come as a separate line, 'image.circle.color = Color.Black' which will make the whole thing more readable